appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -1012,19 +1034,22 @@ private static String getContentType(String name) {
         return contentType;
     }
 
-    private org.eclipse.microprofile.openapi.models.media.Schema createSchema(ApiContext context, Class<?> type) {
+    private org.eclipse.microprofile.openapi.models.media.Schema createSchema(ApiContext context, Class<?> type, Type genericType) {
         org.eclipse.microprofile.openapi.models.media.Schema schema = new SchemaImpl();
         SchemaType schemaType = ModelUtils.getSchemaType(type);
         schema.setType(schemaType);
 
         // Set the subtype if it's an array (for example an array of ints)
         if (schemaType == SchemaType.ARRAY) {
-            Class<?> subType = type.getComponentType();
-            org.eclipse.microprofile.openapi.models.media.Schema subSchema = schema;
-            while (subType != null) {
```

Comment:
FYI: Changed this from looping to recursion (see `schema.setItems` to the right) since the return type could be any combination of collections, arrays and simple types. So this now supports collections is well which required to carry the full generic type through the methods involved.